### PR TITLE
[Concurrency] Change TypeBase::isSendable() to return false for unavailable conformances.

### PR DESCRIPTION
**Pull request by gottesmm**
rdar://122174496

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Review by hborla — APPROVED** [review]

**Comment by gottesmm** [comment]
@swift-ci test

File: lib/AST/ConformanceLookup.cpp [hunk 1]
```diff
@@ -769,7 +769,7 @@ bool TypeBase::isSendableType() {
     return fas->isSendable();
 
   auto conformance = proto->getParentModule()->checkConformance(this, proto);
-  if (conformance.isInvalid())
+  if (conformance.isInvalid() || conformance.hasUnavailableConformance())
     return false;
 
   // Look for missing Sendable conformances.
```
Comment (slavapestov) [inline lib/AST/ConformanceLookup.cpp:775]: The `conformance.forEachMissingConformance(...)` thing is equivalent to `conformance.hasMissingConformances()`. However, you can remove that check entirely by passing `/*allowMissing=*/false` to the checkConformance() call on the previous line.

**Comment by gottesmm** [comment]
@swift-ci test macOS platform

File: lib/AST/ConformanceLookup.cpp (hunk 1 above)
Comment (gottesmm) [reply lib/AST/ConformanceLookup.cpp:775]: I'll do that in a follow on.

**Comment by gottesmm** [comment]
The failures here are two parts:

1. We are now diagnosing a real region isolation case that we weren't handling before!
2. Separately the AST errors around synthesized conformance errors now seem to go down a different diagnostic path and are emitting a different error. The problem is that this error seems to expect to be able to dump a location that a use is but we are emitting errors in a synthesized conformance... so there is nothing to emit, so we get a diagnostic in "elsewhere"

**Comment by gottesmm** [comment]
I did this in a different PR by mistake.

**Pull request closed without merging**